batch/batch/server/server.py

```diff
@@ -145,6 +161,27 @@ def set_state(self, new_state):
                 self._state,
                 new_state))
             self._state = new_state
+            self.notify_children(new_state)
+
+    def notify_children(self, new_state):
+        for child_id in self.children:
```

Comment:
renaming `children` to `child_ids` might make this clearer.  This idiom shows up in a few places.

Reply:
addressed